### ISSUE TITLE
CaffeFunction ignoring scaling factors in BatchNorm layers

### ISSUE THREAD

**Issue by hvy** [chainer/chainer#3260]
Batch normalization is not loaded properly from Caffe models. More precisely, the scaling factor stored as the third item in the blob is not taken into account. I am not sure if this scaling factor is relevant in all Caffe models, but some.

#### Reproduction steps
1. Load a Caffe model that includes a `BathNorm` layer using `CaffeFunction`.
2. Feed the model with an input with `chainer.using_config('train', False)` for inference (corresponding to setting `use_global_stats` to `True` in Caffe).
3. Notice that the the predictions are wrong, caused by either too small or too large means and variances in the batch normalization layers.

#### How Caffe uses the scaling factor

https://github.com/BVLC/caffe/blob/master/src/caffe/layers/batch_norm_layer.cpp#L100-L105

#### Related code in Chaner

https://github.com/chainer/chainer/blob/master/chainer/links/caffe/caffe_function.py#L346-L371

#### Related discussion from the Caffe repository

https://github.com/BVLC/caffe/issues/4885

**Comment by hvy** [comment]
Sent a PR that fixes this issue. https://github.com/chainer/chainer/pull/3261
